src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Controllers/DelegationsController.cs

```diff
@@ -82,6 +85,64 @@ public async Task<ActionResult> Post([FromBody] List<Rule> rules)
             }
         }
 
+        /// <summary>
+        /// Endpoint for retrieving delegated rules between parties
+        /// </summary>
+        /// <response code="400">Bad Request</response>
+        /// <response code="500">Internal Server Error</response>
+        [HttpPost]
+        [Authorize(Policy = AuthzConstants.ALTINNII_AUTHORIZATION)]
+        [Route("authorization/api/v1/[controller]/GetRules")]
+        public async Task<ActionResult<List<Rule>>> GetRules([FromBody] RuleQuery ruleQuery, [FromQuery] bool onlyDirectDelegations = false)
+        {
+            List<int> coveredByPartyIds = new List<int>();
+            List<int> coveredByUserIds = new List<int>();
+            List<int> offeredByPartyIds = new List<int>();
+            List<string> appIds = new List<string>();
+
+            foreach (PolicyMatch policyMatch in ruleQuery.PolicyMatches)
+            {
+                string org = policyMatch.Resource.FirstOrDefault(match => match.Id == XacmlRequestAttribute.OrgAttribute)?.Value;
+                string app = policyMatch.Resource.FirstOrDefault(match => match.Id == XacmlRequestAttribute.AppAttribute)?.Value;
+                if (!string.IsNullOrEmpty(org) && !string.IsNullOrEmpty(app))
+                {
+                    appIds.Add($"{org}/{app}");
+                }
+
+                if (DelegationHelper.TryGetCoveredByPartyIdFromMatch(policyMatch.CoveredBy, out int partyId))
+                {
+                    coveredByPartyIds.Add(partyId);
+                }
+                else if (DelegationHelper.TryGetCoveredByUserIdFromMatch(policyMatch.CoveredBy, out int userId))
+                {
+                    coveredByUserIds.Add(userId);
+                }
+
+                if (ruleQuery.KeyRolePartyIds.Any(id => id != 0))
```

Comment:
This does't seem to be related to the policyMatches your retrieve? Could this be outside of the foreach?

Reply:
Thos goes for parentPartyId as well

Reply:
You're right, I moved them out of the foreach.